### PR TITLE
Escape base and head in repos_commits.go

### PR DESCRIPTION
**Pull request by chughpiyush**
Fixes: #1642 as reported in #1654 

I added code to escape `base` and `head` part of the URL in `CompareCommits` and `CompareCommitsRaw` methods.
Also did changes in relevant test methods, which are now testing for the following 3 cases:

 ```java
         {base: "b", head: "h"},
		{base: "123base", head: "head123"},
		{base: "`~!@#$%^&*()_+-=[]\\{}|;':\",./<>?/*-+123base", head: "head123`~!@#$%^&*()_+-=[]\\{}|;':\",./<>?/*-+"},
```
P.S. - Here's the original [pull request](https://github.com/google/go-github/pull/1649) that had to be reverted.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Very nice, @chughpiyush - thank you.
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
LGTM 👌🏼

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Comment by chughpiyush** [comment]
Thank you both @gmlewis and @wesleimp !

**Pull request merged**
